apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2008,14 +2004,15 @@ func (u *UniterAPI) getRelationAppSettings(canAccess common.AuthFunc, relTag str
 		return nil, common.ErrPerm
 	}
 
-	app, err := u.st.Application(appTag.Id())
+	appName := appTag.Id()
+	_, err = rel.Endpoint(appName)
 	if errors.IsNotFound(err) {
 		return nil, common.ErrPerm
 	} else if err != nil {
 		return nil, errors.Trace(err)
 	}
```

Comment:
Delete _, err = rel.Endpoint(appName) here
And move the error checks to below the rel.ApplicationSettings() call.

Reply:
Oh yeah - good call.